platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/cache/ConcurrentArray.java

```diff
@@ -309,11 +309,15 @@ public StandardFuture<Void> parallelTraverse(Executor executor, Consumer<T> acti
             final T[] array = cur.array;
             final int size = cur.size.get();
             executor.execute(() -> {
-                for (int i = 0; i < size; ++i) {
-                    action.accept(array[i]);
-                }
-                if (count.decrementAndGet() == 0) {
-                    result.complete(null);
+                try {
+                    for (int i = 0; i < size; ++i) {
+                        action.accept(array[i]);
+                    }
+                    if (count.decrementAndGet() == 0) {
+                        result.complete(null);
+                    }
+                } catch (Exception e) {
+                    result.cancelWithError(true, e);
```

Comment:
It appears the first parameter is superfluous as no callback was provided for the `StandardFuture`.